pkg/identityserver/bunstore/user_store.go

```diff
@@ -607,7 +607,7 @@ func (s *userStore) updateUserModel( //nolint:gocyclo
 		case "console_preferences.sort_by":
 			updateConsolePreferences = true
 			consolePreferences.SortBy = pb.ConsolePreferences.GetSortBy()
-		case "console_preferences.tutorials":
+		case "console_preferences.tutorials.seen":
```

Comment:
More or less the same comment left in [this PR](https://github.com/TheThingsNetwork/lorawan-stack/pull/7423).

```suggestion
		case "console_preferences.tutorials":
```

Wouldn't this be enough to update the `Tutorial` subfield of `ConsolePreferences`?

I'm asking this but I didn't validate how the fieldmask is created on the Console side of things.

Reply:
@nicholaspcr see the last to commits for the required changes to shorten the field mask from the console side.

By default we don't set the field masks they are derived from the data sent by this function `Marshaler.fieldMaskFromPatch` so thats why it can be confusing to know what field mask is expected without reading the go code when it is developer preference when not to use the default.

I'm leaning towards lets do this on the backend instead of the frontend @ryaplots what do you think?

Reply:
> It could be that it is too annoying to make it so every subfield has its fieldmask shortened to its parent's fieldmask, something like console_preferences.tutorials.seen -> console_preferences.tutorials.

Like Michael said, we have this `Marshaler.fieldMaskFromPatch` function that derives the fieldmasks from the patch. To make Nicholas' suggestion work we would need to refactor this function to default to the parent fieldmask, however, I get the impression that that would not work for every case, because I believe there are fields that expect a more specific fieldmask in the backend, so we would have to adapt this specific fieldmask, which I don't should happen. It's better if the fieldmasks are streamlined.
 
So, I agree with adding `console_preferences.tutorials.seen` and solving this in the backend.

Reply:
I see.

Indeed it makes more sense to have it handled on the backend, specially since this is the standard and it is not hard to do it there.

<hr/>

The suggestion I would make regarding the backend changes would be to have the logic handled updates of subfields.

```suggestion
		case "console_preferences.tutorials":
			updateConsolePreferences = true
			consolePreferences.Tutorials = pb.ConsolePreferences.GetTutorials()
		case "console_preferences.tutorials.seen":
			updateConsolePreferences = true
			consolePreferences.Tutorials.Seen = pb.ConsolePreferences.GetTutorials().GetSeen()
```

@mjamescompton you let me know if you want to do this or if I should make a separate PR adding this to all `UserConsolePreferences` fields.